vars/transportRequestGetOrCreate.groovy

```diff
@@ -0,0 +1,78 @@
+import groovy.transform.Field
+
+import com.sap.piper.ConfigurationMerger
+import com.sap.piper.cm.ChangeManagement
+import com.sap.piper.cm.ChangeManagementException
+
+import hudson.AbortException
+
+
+@Field def STEP_NAME = 'transportRequestGetOrCreate'
+
+@Field Set parameterKeys = [
+    'changeDocumentId',
+    'developmentSystemId',
+    'credentialsId',
+    'endpoint'
+  ]
+
+@Field Set stepConfigurationKeys = [
+    'credentialsId',
+    'endpoint'
+  ]
+
+def call(parameters = [:]) {
+
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
+
+        def script = parameters?.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
+
+        ChangeManagement cm = new ChangeManagement(script)
+
+        Map configuration = ConfigurationMerger.merge(parameters.script, STEP_NAME,
+                                                      parameters, parameterKeys,
+                                                      stepConfigurationKeys)
+
+        def changeDocumentId = configuration.changeDocumentId
+        if(!changeDocumentId) throw new AbortException('Change id not provided (parameter: \'changeDocumentId\').')
+
+        def developmentSystemId = configuration.developmentSystemId
+        if(!developmentSystemId) throw new AbortException('Development system id not provided (parameter: \'developmentSystemId\').')
+
+        def credentialsId = configuration.credentialsId
+        if(!credentialsId) throw new AbortException('Credentials id not provided (parameter: \'credentialsId\').')
+
+        def endpoint = configuration.endpoint
+        if(!endpoint) throw new AbortException('Solution Manager endpoint not provided (parameter: \'endpoint\').')
+
+        def transportRequests
+
+        echo "[INFO] Getting transport requests for change document '$changeDocumentId'."
+
+        withCredentials([usernamePassword(
+            credentialsId: credentialsId,
+            passwordVariable: 'password',
+            usernameVariable: 'username')]) {
+
+            try {
+                transportRequests = cm.getTransportRequests(changeDocumentId, endpoint, username, password)
+            } catch(ChangeManagementException ex) {
+                throw new AbortException(ex.getMessage())
+            }
+
+            if(transportRequests.length>1){
```

Comment:
We should check how this fits into the appraoch with providing the transportId somewhere in the commit history. What happens if 
* we have multipe transports, but there is a transport denoted in the commit message. In case we find this, should we return this, even if there are multiple open transport requests?

* one transport, but this transport does not match with a transport request id provided via commit messages?

Would be also technical possible to ignore transport requests from the commit history in this case. But despite this is technically possible I doubt it would be a cool idea.

Side note: a pipeline step which checks if a (open) transport for a given transport ID exists on the server sounds very reasonable to me (more the less the same like checking if a change represented by a give change document id is in status 'in development' (.. which comes also with an implicit existence check))